### PR TITLE
Add Sarus test for mpi+cuda functionality

### PR DESCRIPTION
**Pull request by teojgo**

### CODE REVIEW

File: checks/mixins/sarus_extra_launcher_options.py
Comment (jgphpc) [inline checks/mixins/sarus_extra_launcher_options.py:12]: Not great to have `hohgant` hardcoded here + the flag  does not seem to be used: see rfm_job.sh of check_gpu_ids.py for example

File: checks/containers/sarus/check_cuda_nbody.py
Comment (jgphpc) [inline checks/containers/sarus/check_cuda_nbody.py:46]: remove `maintainers = ['amadonna', 'taliaga']`

File: checks/containers/sarus/check_cuda_nbody.py
Comment (jgphpc) [inline checks/containers/sarus/check_cuda_nbody.py:5]:
```suggestion
# https://catalog.ngc.nvidia.com/orgs/nvidia/teams/k8s/containers/cuda-sample/tags
```

File: checks/mixins/sarus_extra_launcher_options.py
Comment (teojgo) [reply checks/mixins/sarus_extra_launcher_options.py:12]: On check_gpu_ids, it's not used because it's not an mpi job.

File: checks/containers/sarus/check_cuda_nbody.py
Comment (teojgo) [reply checks/containers/sarus/check_cuda_nbody.py:5]: ?

File: checks/containers/sarus/check_pyfr.py
Comment (jgphpc) [inline checks/containers/sarus/check_pyfr.py:5]: add 
```
prerun_cmds += ['unset XDG_RUNTIME_DIR']
```
or
```
self.env_vars['XDG_RUNTIME_DIR'] = ''
```

File: checks/containers/sarus/check_gpu_ids.py
Comment (jgphpc) [inline checks/containers/sarus/check_gpu_ids.py:14]: remove ?

Comment (teojgo) [reply checks/containers/sarus/check_gpu_ids.py:14]: It's needed

File: checks/containers/sarus/check_osu_benchmarks.py
Comment (jgphpc) [inline checks/containers/sarus/check_osu_benchmarks.py:1]: remove double space @ line 67:  `self.container_platform.command  = (`

File: checks/containers/sarus/check_pyfr.py
Comment (teojgo) [reply checks/containers/sarus/check_pyfr.py:5]: It's done in the setup container platform

File: checks/containers/sarus/check_osu_benchmarks.py
Comment (jgphpc) [inline checks/containers/sarus/check_osu_benchmarks.py:17]: you may want to move to a more recent version but for the sake of time, let's skip updating the version.

Comment (teojgo) [reply checks/containers/sarus/check_osu_benchmarks.py:17]: This is a matter of the image that we are using

**Review by jgphpc — APPROVED** [review]
minor changes needed but lgtm

File: checks/containers/sarus/check_osu_benchmarks_cuda.py
Comment (jgphpc) [inline checks/containers/sarus/check_osu_benchmarks_cuda.py:42]:
```suggestion
        self.container_platform.command = (
```

**Pull request merged**
